### ISSUE TITLE
Upgrade nuget-client on CI

### ISSUE THREAD

**Issue by apacha** [bitrise-steplib/steps-nuget-restore#5]
When trying to restore my nuget-package on Bitrise, it always fails with the following log-output:

```
/Library/Frameworks/Mono.framework/Versions/Current/bin/nuget restore Source/MySolution.sln
Installing 'Xamarin.Android.Support.v4 23.0.1.3'.
Installing 'WPtoolkit 4.2013.08.16'.
Installing 'Xamarin.Android.Support.v7.MediaRouter 23.0.1.3'.
Installing 'NUnit 2.6.4'.
Installing 'Xamarin.Android.Support.Design 23.0.1.3'.
Installing 'MvvmCross.Platform 4.0.0'.
Installing 'MvvmCross.PortableSupport 3.5.1'.
Installing 'MvvmCross.Core 4.0.0'.
Installing 'Xamarin.Android.Support.v7.CardView 23.0.1.3'.
Installing 'Xamarin.Forms 2.0.1.6505'.
Installing 'MvvmCross 4.0.0'.
Installing 'NSubstitute 1.9.2.0'.
Installing 'Xamarin.Android.Support.v7.AppCompat 23.0.1.3'.
Installing 'MvvmCross.Binding 4.0.0'.
Installing 'MvvmCross.Forms.Presenter 4.0.1'.
Successfully installed 'Xamarin.Android.Support.v7.MediaRouter 23.0.1.3'.
Successfully installed 'Xamarin.Android.Support.v7.CardView 23.0.1.3'.
Successfully installed 'MvvmCross.PortableSupport 3.5.1'.
Successfully installed 'Xamarin.Android.Support.v7.AppCompat 23.0.1.3'.
Successfully installed 'Xamarin.Android.Support.v4 23.0.1.3'.
Successfully installed 'Xamarin.Android.Support.Design 23.0.1.3'.
Successfully installed 'NUnit 2.6.4'.
Successfully installed 'NSubstitute 1.9.2.0'.
Successfully installed 'MvvmCross.Platform 4.0.0'.
Successfully installed 'WPtoolkit 4.2013.08.16'.
Successfully installed 'Xamarin.Forms 2.0.1.6505'.
'MvvmCross.Forms.Presenter' already has a dependency defined for 'MvvmCross.Core'.
'MvvmCross' already has a dependency defined for 'MvvmCross.Core'.
'MvvmCross.Binding' already has a dependency defined for 'MvvmCross.Core'.
'MvvmCross.Core' already has a dependency defined for 'MvvmCross.Platform'.
ERRO[10:10:10] Step (nuget-restore) failed, error: (exit status 1) 
```

When running the same process on any machine locally (Windows or Mac), it works as expected and the packages are restored properly. 
I found this [thread](https://github.com/MvvmCross/MvvmCross/issues/1088) and [this one](https://github.com/MvvmCross/MvvmCross-Forms/issues/23) and they clearly indicate that it is a problem with an old nuget version. I'm not sure if the Xamarin Version on the CI is responsible for this, of if the nuget-version can be changed independently. Maybe an upgrade could fix this.

**Comment by viktorbenei** [comment]
Hi, you can find a guide here about how to use a custom NuGet version: http://devcenter.bitrise.io/docs/nuget-restore-with-custom-nuget-version - let us know if that wouldn't help!

About the pre-installed NuGet version: we pre-install the version which comes with Xamarin. If they decide to use a newer version it'll be pre-installed on the VMs too.

**Comment by viktorbenei** [comment]
I'll close this ticket, but feel free to add a comment if the custom nuget version solution wouldn't work for you, and we'll re-open the ticket.

**Comment by apacha** [comment]
I've tried to create a custom script instead of the predefined nuget-restore with the code provided by you and replaced version 2.8.6 with 3.3.0 and now the NuGet restore is simply hanging in my build forever...

```
+------------------------------------------------------------------------------+
| (4) Custom Nuget restore                                                     |
+------------------------------------------------------------------------------+
| id: script                                                                   |
| version: 1.1.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| time: 2016-02-22T01:43:09-08:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |
Downloading NuGet version 3.3.0
NuGet Restore
MSBuild auto-detection: using msbuild version '4.0' from '/Library/Frameworks/Mono.framework/Versions/4.2.2/lib/mono/4.5'.
Feeds used:
  https://api.nuget.org/v3/index.json
Restoring NuGet package MvvmCross.Forms.Presenter.4.0.1.
Restoring NuGet package MvvmCross.4.0.0.
Restoring NuGet package MvvmCross.Binding.4.0.0.
Restoring NuGet package MvvmCross.Core.4.0.0.
Restoring NuGet package MvvmCross.Platform.4.0.0.
Restoring NuGet package MvvmCross.PortableSupport.3.5.1.
Restoring NuGet package Xamarin.Forms.2.0.1.6505.
Restoring NuGet package Xamarin.Android.Support.Design.23.0.1.3.
Restoring NuGet package Xamarin.Android.Support.v4.23.0.1.3.
Restoring NuGet package Xamarin.Android.Support.v7.AppCompat.23.0.1.3.
Restoring NuGet package Xamarin.Android.Support.v7.CardView.23.0.1.3.
Restoring NuGet package Xamarin.Android.Support.v7.MediaRouter.23.0.1.3.
Restoring NuGet package WPtoolkit.4.2013.8.16.
Restoring NuGet package NSubstitute.1.9.2.
Restoring NuGet package NUnit.2.6.4.
```

**Comment by viktorbenei** [comment]
@apacha can you please invite our support account to your app on bitrise.io?

Support account username: `bitrise-support`

**Comment by apacha** [comment]
Done.

**Comment by viktorbenei** [comment]
Thanks, we'll look into it ASAP!

**Comment by birmacher** [comment]
Hi @apacha!

It seems that version `3.3.0` hangs in some cases. I changed the NuGet version from `3.3.0` to `2.8.6` and everything is working great now.
( Xamarin is shipped with `2.8.5` )

I'm closing the ticket because the issue is resolved, but feel free to update it if you hit any issues.
Also I found a couple issues with NuGet restore, like: https://github.com/NuGet/Home/issues/2150
It seems that it's not just VS related.
